File: 백준/Bronze/21756. 지우개/지우개.py
n = int(input())

vec = [i+1 for i in range(n)]

while len(vec) > 1:
  count = 0
  l = len(vec)
  for k in range(l):
    if k%2 == 0:
      vec.pop(k-count)
      count += 1

print(*vec)

File: 백준/Bronze/21756. 지우개/지우개.cc
#include <iostream>
#include <vector>
using namespace std;

int main() {
  int n;
  cin >> n;

  vector<int> vec;
  for(int i=1; i<n+1; i++) vec.push_back(i);

  while(vec.size() > 1) {
    int count = 0, l;
    l = vec.size();
    for (int k=0; k<l; k++) {
      if (k%2 == 0) {
        vec.erase(vec.begin()+k-count);
        count ++;
      }
    }
  }
  cout << vec[0] << endl;
  
  return 0;
}